torch/distributed/distributed_c10d.py

```diff
@@ -180,8 +180,8 @@ def _get_group_rank(group, rank):
         raise RuntimeError("The given group does not exist")
     try:
         group_rank = _pg_group_ranks[group][rank]
-    except KeyError:
-        raise RuntimeError("The global rank is not part of the group")
+    except KeyError as e:
+        raise RuntimeError("The global rank is not part of the group") from e
```

Comment:
KeyError doesn't convey much additional info to the developer. Perhaps just adding additional info the the error message would be better?
```suggestion
    except KeyError:
        raise RuntimeError(f"The global rank {rank} is not part of the group {group}") from None
```

Reply:
I agree.